app/src/main/java/com/asfoundation/wallet/subscriptions/SubscriptionCancelPresenter.kt

```diff
@@ -0,0 +1,84 @@
+package com.asfoundation.wallet.subscriptions
+
+import com.asfoundation.wallet.util.isNoNetworkException
+import io.reactivex.Scheduler
+import io.reactivex.disposables.CompositeDisposable
+import java.util.concurrent.TimeUnit
+
+class SubscriptionCancelPresenter(
+    private val subscriptionInteract: SubscriptionInteract,
+    private val disposables: CompositeDisposable,
+    private val networkScheduler: Scheduler,
+    private val viewScheduler: Scheduler,
+    private val view: SubscriptionCancelView
+) {
+
+  fun present(packageName: String) {
+    loadSubscriptionDetails(packageName)
+    handleCancelClicks()
+    handleBackClicks()
+  }
+
+  private fun loadSubscriptionDetails(packageName: String) {
+    disposables.add(
+        subscriptionInteract.loadSubscriptionDetails(packageName)
+            .delay(1, TimeUnit.SECONDS)
```

Comment:
why the delay?

Reply:
it was only for mocked purposes since the view is displaying in the same instant